### PR TITLE
[SYCL][UR][L0] Correctly set OwnNativeHandle on image create

### PR DESCRIPTION
**Pull request by jandres742**

### CODE REVIEW

File: sycl/plugins/unified_runtime/ur/adapters/level_zero/memory.cpp [hunk 1]
```diff
@@ -1621,7 +1621,7 @@ UR_APIEXPORT ur_result_t UR_APICALL urMemImageCreate(
              (Context->ZeContext, Device->ZeDevice, &ZeImageDesc, &ZeImage));
 
   try {
-    auto UrImage = new _ur_image(Context, ZeImage);
+    auto UrImage = new _ur_image(Context, ZeImage, true /*OwnZeMemHandle*/);
```
Comment (smaslov-intel) [inline sycl/plugins/unified_runtime/ur/adapters/level_zero/memory.cpp:1624]: Perhaps better not to provide a default value of "false"

**Review by smaslov-intel — APPROVED** [review]

File: sycl/plugins/unified_runtime/ur/adapters/level_zero/memory.cpp (hunk 1 above)
Comment (jandres742) [reply sycl/plugins/unified_runtime/ur/adapters/level_zero/memory.cpp:1624]: possibly, but the way we have it is OwnNativeHandle is defined in _ur_object here https://github.com/intel/llvm/blob/b9c8516ddd1195501a5914f09c84d7d3f83318a1/sycl/plugins/unified_runtime/ur/adapters/level_zero/common.hpp#L388, so changing that default would affect more than just images. Changing that default could be more risky, so prefer to have it as is now to have a more contained change.

**Comment by jandres742** [comment]
error is with esimd:gpu but changes on this patch are specific to level-zero:gpu, so failure is unrelated

```sh
$ "env" "CM_RT_PLATFORM=skl" "ONEAPI_DEVICE_SELECTOR=ext_intel_esimd_emulator:gpu" "/__w/llvm/llvm/build-e2e/ESIMD/Output/acc_gather_scatter_rgba.cpp.tmp.out"
note: command had no output on stdout or stderr
error: command failed with exit status: -11
```

File: sycl/plugins/unified_runtime/ur/adapters/level_zero/memory.cpp (hunk 1 above)
Comment (smaslov-intel) [reply sycl/plugins/unified_runtime/ur/adapters/level_zero/memory.cpp:1624]: That is OK. If we ever refactor this then perhaps the better default would "true".

**Comment by jandres742** [comment]
@intel/llvm-gatekeepers : please merge when possible.

**Comment by aelovikov-intel** [comment]
> error is with esimd:gpu but changes on this patch are specific to level-zero:gpu, so failure is unrelated

And we're removing that target anyway.

**Pull request merged**